### PR TITLE
Make all Scala body parsers `maxLength: Long` (#8519)

### PR DESCRIPTION
**Pull request by yawaramin**
The Java body parsers already uniformly take `long maxLength` (see
comment in issue).

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @yawaramin,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by yawaramin** [comment]
Binary compatibility check failed, which makes sense. If this is going to go in, it can't be before Play 3, I guess.

**Comment by marcospereira** [comment]
Hi @yawaramin,

It can be Play 2.7. Play is not following semantic versioning yet (there are plans to adopt semver after Play 3). So, what you need to do is to run the following from the project root:

```bash
cd framework && sbt mimaReportBinaryIssues
```
See where the change is breaking binary compatibility and add the filters here:

https://github.com/playframework/playframework/blob/426d0b23af3b45b97b3c13aaa8432817e62b9d56/framework/project/BuildSettings.scala#L181

Best.

**Comment by yawaramin** [comment]
@marcospereira thanks! I've updated as per your comment.

**Comment by marcospereira** [comment]
Hi @yawaramin,

I was looking at the Java API when this wild cast appears:

https://github.com/playframework/playframework/blob/aaf166be62b3ab27b88eafc0b18e8dab0acbe367/framework/src/play/src/main/java/play/mvc/BodyParser.java#L516

`maxLength` is a long and the cast can result in an overflow when its value is greater than `Integer.MAX_VALUE` (should be unusual, but it is possible). I think we should also use a `Long` in `Multipart.multipartParser`:

https://github.com/playframework/playframework/blob/c038bc147b03285a1de7acc50037e1e7f438bedf/framework/src/play/src/main/scala/play/core/parsers/Multipart.scala#L80

`Multipart.multipartParser` is also used here:

https://github.com/playframework/playframework/blob/a39b208659fbb79f179a795331805fab9cc132a1/framework/src/play/src/main/scala/play/api/mvc/BodyParsers.scala#L848

Which means we also want to change `DefaultMaxTextLength` to be a `Long`. I'm not so sure about this last part since having it as an `Int` helps to have a up limit to the amount of data will be in memory. Typesafe Config will throw an exception if the configuration is out of integer bounds.

@schmitch, what do you think?

**Comment by schmitch** [comment]
Well I'm in for changing all maxLength to `Long`. 
I've actually had problems with these Int's already.
I mean considering that Max Length is a byte which means that if I want to express 10gb: `1 * 1024 * 1024 * 1024 * 10` it would already overflow. (not that this is a common use case, because having a 10gb heap is already insane...)

File: framework/project/BuildSettings.scala [hunk 1]
```diff
@@ -555,7 +555,25 @@ object BuildSettings {
 
       // Remove DefaultDBApi.connect method
       ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.db.DefaultDBApi.connect$default$1"),
-      ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.db.DefaultDBApi.connect")
+      ProblemFilters.exclude[DirectMissingMethodProblem]("play.api.db.DefaultDBApi.connect"),
```
Comment (rakeshhny) [inline framework/project/BuildSettings.scala]: Removed and added same thing again..?

Comment (marcospereira) [reply framework/project/BuildSettings.scala]: That is because there is a conflict at the current state of PR. :-)

**Comment by marcospereira** [comment]
Hey @yawaramin,

There is a conflict here now. Let me know if you need some help with that and also with the points at [this comment](https://github.com/playframework/playframework/pull/8520#issuecomment-405956897).

**Pull request merged**